src/robotFactory.js

```diff
@@ -1,10 +1,71 @@
+/* eslint-disable no-console */
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
```suggestion
    this.chipVersion = chipVersion;
    this.coords = { 
      x: coords.x || 0,
      y: coords.y || 0,
    }
```
You can do it like this